packages/faceted-search/src/components/Badges/BadgeCheckboxes/BadgeCheckboxesForm.component.js

```diff
@@ -55,13 +48,17 @@ const createCheckboxEntity = value => checkbox => {
 	};
 };
 
-const getCheckboxes = (checkboxes, value, filterValue) =>
-	checkboxes
-		.filter(checkbox => checkbox.label.toLowerCase().includes(filterValue))
+const getCheckboxes = (checkboxes, value, filterValue) => {
+	const formatFilterValue = filterValue.trim().toLowerCase();
+
+	return checkboxes
+		.filter(checkbox => checkbox.label.toLowerCase().includes(formatFilterValue))
 		.map(createCheckboxEntity(value));
+};
 
 const BadgeCheckboxesForm = ({ checkboxValues, id, onChange, onSubmit, value, t }) => {
-	const [filter, onFilter, resetFilter] = useFilter();
+	const [filter, setFilter] = useState('');
+
 	const badgeCheckBoxesFormId = `${id}-checkboxes-form`;
 	const checkboxes = useMemo(() => getCheckboxes(checkboxValues, value, filter), [
```

Comment:
We can consider using `useCallback` instead

Reply:
You're right, i make the change